php/class-plugin-base.php

```diff
@@ -152,17 +152,7 @@ public function locate_plugin() {
 		if ( 0 === $count ) {
 			throw new \Exception( "Class not located within a directory tree containing 'plugins': $file_name" );
 		}
-
-		// Make sure that we can reliably get the relative path inside of the content directory.
-		$content_dir = realpath( trailingslashit( WP_CONTENT_DIR ) );
-		if ( '/' !== \DIRECTORY_SEPARATOR ) {
-			$content_dir = str_replace( \DIRECTORY_SEPARATOR, '/', $content_dir ); // Windows compat.
-		}
-		if ( 0 !== strpos( $plugin_dir, $content_dir ) ) {
-			throw new \Exception( 'Plugin dir is not inside of WP_CONTENT_DIR' );
-		}
-		$content_sub_path = substr( $plugin_dir, strlen( $content_dir ) );
-		$dir_url = content_url( trailingslashit( $content_sub_path ) );
+		$dir_url = content_url( trailingslashit( 'plugins/' . basename( $plugin_dir ) ) );
```

Comment:
@MattGeri how about adding an `if`/`else` in here to check whether or not `defined( 'WPCOM_IS_VIP_ENV' )`. And if so, run the code you deleted, but if not VIP, then use the new line you added. Nevertheless, in that case, should we just use `plugin_dir_url()`?